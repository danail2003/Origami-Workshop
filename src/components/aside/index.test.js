import React from 'react';
import renderer from 'react-test-renderer';
import Aside from './index';
import UserContext from '../../context';
import { Route, BrowserRouter } from 'react-router-dom';

describe('Aside', () => {
    it('should render authenticated routes', () => {
        const component = renderer.create(
            <BrowserRouter>
                <Route>
                    <UserContext.Provider value={{
                        user: {
                            loggedIn: true,
                            id: '456'
                        }
                    }}>
                        <Aside />
                    </UserContext.Provider>
                </Route>
            </BrowserRouter>
        ).toJSON();
        expect(component).toMatchSnapshot()
    });

    it('should render non-authenticated routes', () => {
        const component = renderer.create(
            <BrowserRouter>
                <Route>
                    <UserContext.Provider value={{
                        user: {
                            loggedIn: false
                        }
                    }}>
                        <Aside />
                    </UserContext.Provider>
                </Route>
            </BrowserRouter>
        ).toJSON();
        expect(component).toMatchSnapshot()
    });
});