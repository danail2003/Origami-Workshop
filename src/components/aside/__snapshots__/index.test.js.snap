// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aside should render authenticated routes 1`] = `
<aside
  className="aside"
>
  <ul>
    <li
      className="aside-list-item"
    >
      <a
        className="aside-link"
        href="/"
        onClick={[Function]}
      >
        Publications
      </a>
    </li>
    <li
      className="aside-list-item"
    >
      <a
        className="aside-link"
        href="/share"
        onClick={[Function]}
      >
        Share your thoughts
      </a>
    </li>
    <li
      className="aside-list-item"
    >
      <a
        className="aside-link"
        href="/profile/456"
        onClick={[Function]}
      >
        Profile
      </a>
    </li>
  </ul>
</aside>
`;

exports[`Aside should render non-authenticated routes 1`] = `
<aside
  className="aside"
>
  <ul>
    <li
      className="aside-list-item"
    >
      <a
        className="aside-link"
        href="/"
        onClick={[Function]}
      >
        Publications
      </a>
    </li>
    <li
      className="aside-list-item"
    >
      <a
        className="aside-link"
        href="/register"
        onClick={[Function]}
      >
        Register
      </a>
    </li>
    <li
      className="aside-list-item"
    >
      <a
        className="aside-link"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </li>
  </ul>
</aside>
`;
